docs/adr/0004-runtime-governance.md

```diff
@@ -0,0 +1,225 @@
+
+The runtime descriptor version is bumped to `1`. Version `0` descriptors are
+accepted at genesis and are converted to the new format by assuming the entity
+governance model as that is the only option in v0. All new runtime registrations
+must use the v1 descriptor.
+
+This proposal updates the runtime descriptor by adding fields as follows:
+
+```golang
+type Runtime struct {
+    GovernanceModel RuntimeGovernanceModel `json:"governance_model"`
+
+    // ... existing fields omitted ...
+}
+
+// RuntimeGovernanceModel specifies the runtime governance model.
+type RuntimeGovernanceModel uint8
+
+const (
+    GovernanceEntity    RuntimeGovernanceModel = 1
+    GovernanceRuntime   RuntimeGovernanceModel = 2
+    GovernanceConsensus RuntimeGovernanceModel = 3
+)
+
+// ... some text serialization methods omitted ...
+```
+
+The `governance_model` field can specifiy one of the following governance
+models:
+
+- **Entity governance (`GovernanceEntity`).** This causes the runtime to behave
+  exactly as before, the runtime owner (indicated by `entity_id` in the runtime
+  descriptor) is the only one who can update the runtime descriptor via
+  `registry.RegisterRuntime` method calls subject to the `runtime_update_delay`
+  registry consensus parameter.
+
+  The runtime owner is also the one that needs to provide the required stake
+  in escrow in order to avoid the runtime from being suspended. As before note
+  that anyone can delegate the required stake to the runtime owner in order to
+  enable runtime operation (but the owner can always prevent the runtime from
+  operating by performing actions which would cause the stake claims to no
+  longer be satisfied).
+
+- **Runtime-defined governance (`GovernanceRuntime`).** In this case the runtime
+  itself is the only one who can update the runtime descriptor by emitting a
+  runtime message subject to the `runtime_update_delay` registry consensus
+  parameter. The runtime owner (indicated by `entity_id`) is not able to perform
+  any updates after the initial registration and such attempts must return
+  `ErrForbidden`.
+
+  The runtime itself is the one that needs to provide the required stake in
+  escrow in order to avoid the runtime from being suspended. This assumes that
+  runtimes can have accounts in the staking module as specified by [ADR 0003].
+  Note that anyone can delegate the required stake to a runtime in order to
+  enable its operation.
+
+- **Consensus layer governance (`GovernanceConsensus`).** In this case only the
+  consensus layer itself can update the runtime descriptor either through a
+  network upgrade or via a consensus layer governance mechanism not specified by
+  this proposal.
+
+  Runtimes using this governance model are never suspended and do not need to
+  provide stake in escrow.
+
+  Runtimes using this governance model cannot be registered via regular registry
+  method calls (doing so must return `ErrForbidden`). Instead such a runtime can
```

Comment:
I think this includes when an owning entity or runtime tries to update its own descriptor, is that right?

Reply:
Correct.